### DISCUSSION TITLE
failed to load initrd generated from update-initramfs tool

### DISCUSSION THREAD

**Discussion by leslie-qiwa** [ipxe/ipxe#1251] [General]
I'm trying to log into a ramfs shell with ssh using ipxe. This is what I did:
1. Generate a initrd.img from ubuntu 22.04 by following steps
 - apt install dropbear-initramfs
 - update-initramfs -u
 - copy /boot/vmlinuz and /boot/initrd.img to iPXE server 
2. Below is one newly added menu in install.ipxe file
```
:UBUNTU22.04
kernel vmlinuz
initrd initrd.img
boot || goto failed
```
3. From kvm console, Ipxe can load both kernel and initrd file, but it get into below failure soon.
/dev/root: Can not open blockdev
VFS: can not open root device (null)

![image](https://github.com/ipxe/ipxe/assets/34456213/169f8334-ef2e-401a-abe6-1e11fd52e14e)

4. The same file works under legacy bios PXE boot using pxelinux.0

Can someone please suggest what could be wrong?

**Comment by NiKiZe** [comment]
Which version of iPXE (see the bootup banner), and which version is the kernel? (Use `uname`)

Earlier versions of either requires you to add kernel cmdline to tell it which file in efi filesystem that should be used by the kernel for initramfs.

You can also search for the "kernel panic - not syncing: VFS" to get more complete explanations and fixes.
